ui/components/app/modals/add-new-contact-modal/add-new-contact-modal.component.js

```diff
@@ -0,0 +1,160 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import { debounce } from 'lodash';
+import { isValidAddress } from 'ethereumjs-util';
+import { providers } from 'ethers';
+import Identicon from '../../../ui/identicon';
+import TextField from '../../../ui/text-field';
+import { isValidDomainName } from '../../../../helpers/utils/util';
+import PageContainerFooter from '../../../ui/page-container/page-container-footer';
+import { getEnvironmentType } from '../../../../../app/scripts/lib/util';
+import { ENVIRONMENT_TYPE_FULLSCREEN } from '../../../../../shared/constants/app';
+import { isBurnAddress } from '../../../../../shared/modules/hexstring-utils';
+
+const environmentType = getEnvironmentType();
+const isFullScreen = environmentType === ENVIRONMENT_TYPE_FULLSCREEN;
+export default class AddNewContactModal extends PureComponent {
```

Comment:
[REQ] in addition to moving this from modals and using a popover, all new components we create should not use the .component / .container breakdown and instead use functional components that use the useSelector/useDispatch hooks to tie into redux stores